test/archethic/p2p/client/connection_test.exs

```diff
@@ -45,6 +45,36 @@ defmodule Archethic.P2P.Client.ConnectionTest do
               }} = :sys.get_state(pid)
     end
 
+    test "should get an error, :closed when trying to reach an unreachable node" do
+      defmodule MockTransportUnreachable do
+        alias Archethic.P2P.Client.Transport
+
+        @behaviour Transport
+
+        def handle_connect(_ip, _port) do
+          {:error, :timeout}
```

Comment:
Here you should add `Process.sleep` to simulate a long connection time.
Actually the `assert {:error, :closed}` may return the result after the connection is done event with `Process.sleep`
The test should test if we receive a return `{:error, :closed}` before the connect function is finished